pkg/devserver/devserver.go

```diff
@@ -295,6 +296,7 @@ func start(ctx context.Context, opts StartOpts) error {
 		Continuations: true,
 	}
 	enableKeyQueues := os.Getenv("EXPERIMENTAL_KEY_QUEUES_ENABLE") == "true"
+	enableStepMetadata := os.Getenv("EXPERIMENTAL_STEP_METADATA") == "true"
```

Comment:
I think this is good for the first merge to be safe, but noting that we should be able to fast follow by always enabling this in local dev - Cloud can still be protected though.